frameworks/shared/serialization.py

```diff
@@ -53,36 +112,45 @@ def serialize_data(data, path):
             data.to_hdf(path, os.path.basename(path), mode='w', format='table')
         elif ser == 'json':
             data.to_json(path, compression=ser_config.pandas_compression)
-    elif np:
-        root, ext = os.path.splitext(path)
-        path = f"{root}.npy"
-        np.save(path, data, allow_pickle=True)
     else:
-        raise ImportError("Numpy or Pandas are required to serialize data between processes.")
+        raise ImportError(f"Numpy or Pandas are required to serialize data between processes to {path}.")
```

Comment:
I would suggest making this error message more specific based on data type:

 > Package <numpy/pandas/scipy> is required to serialize data of type {type(data)} to {path}.

If not adding the specific package, then for developers at least the data type will help deduce which package was missing. Otherwise it is not immediately clear [what went wrong](https://github.com/openml/automlbenchmark/pull/371/checks?check_run_id=3186600314#step:7:741).

Reply:
well, if I have a object of type `ndarray` then I obviously have `numpy` loaded, same with pandas and so on… so the only use-case is when there's nothing… which should never happen.
anyway, I decided to fallback to `pickle` or `json`, which will be able to serialize only simple data structures without additional dependencies.